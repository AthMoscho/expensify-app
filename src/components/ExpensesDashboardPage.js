import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListItem from './ExpenseListItem';
import ExpenseListFilters from './ExpenseListFilters';

const ExpensesDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpensesDashboardPage;