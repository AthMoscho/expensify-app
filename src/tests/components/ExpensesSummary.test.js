import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import numeral from 'numeral';

numeral.register('locale', 'gr', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    currency: {
        symbol: 'â‚¬'
    }
})

numeral.locale('gr');

test('Should render expensesSummary with 1 expense' ,() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={55203}/>)
    expect(wrapper).toMatchSnapshot();
})


test('Should render expensesSummary with multiple expenses' ,() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={55222503}/>)
    expect(wrapper).toMatchSnapshot();
})