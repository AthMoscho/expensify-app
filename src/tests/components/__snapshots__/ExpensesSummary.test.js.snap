// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        €552.03
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render expensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totaling 
      <span>
        €552,225.03
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
